diff --git a/src/Plugin/GraphQL/Schemas/SchemaPluginBase.php b/src/Plugin/GraphQL/Schemas/SchemaPluginBase.php
index dc60dd9..1c2bd89 100644
--- a/src/Plugin/GraphQL/Schemas/SchemaPluginBase.php
+++ b/src/Plugin/GraphQL/Schemas/SchemaPluginBase.php
@@ -439,7 +439,15 @@ abstract class SchemaPluginBase extends PluginBase implements SchemaPluginInterf
    * {@inheritdoc}
    */
   public function processFields(array $fields) {
-    return array_map([$this, 'buildField'], $fields);
+    $processFields = array_map([$this, 'buildField'], $fields);
+
+    foreach($processFields as $key => $processField) {
+      if ($processField === false) {
+        unset($processFields[$key]);
+      }
+    }
+
+    return $processFields;
   }
 
   /**
@@ -506,6 +514,10 @@ abstract class SchemaPluginBase extends PluginBase implements SchemaPluginInterf
    *   The field definition.
    */
   protected function buildField($field) {
+    if ($field['definition']['type'][0] === 'layout_section') {
+      return false;
+    }
+
     if (!isset($this->fields[$field['id']])) {
       $creator = [$field['class'], 'createInstance'];
       $this->fields[$field['id']] = $creator($this, $this->fieldManager, $field['definition'], $field['id']);
