Subject: [PATCH] 2942404-messages-should-have-role-status
---
Index: modules/big_pipe/tests/modules/big_pipe_test/src/BigPipePlaceholderTestCases.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/big_pipe/tests/modules/big_pipe_test/src/BigPipePlaceholderTestCases.php b/modules/big_pipe/tests/modules/big_pipe_test/src/BigPipePlaceholderTestCases.php
--- a/modules/big_pipe/tests/modules/big_pipe_test/src/BigPipePlaceholderTestCases.php	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/modules/big_pipe/tests/modules/big_pipe_test/src/BigPipePlaceholderTestCases.php	(date 1688139352024)
@@ -95,11 +95,11 @@
           'command' => 'insert',
           'method' => 'replaceWith',
           'selector' => '[data-big-pipe-placeholder-id="callback=Drupal%5CCore%5CRender%5CElement%5CStatusMessages%3A%3ArenderMessages&args%5B0%5D&token=_HAdUpwWmet0TOTe2PSiJuMntExoshbm1kh2wQzzzAA"]',
-          'data' => '<div data-drupal-messages>' . "\n" . ' <div role="contentinfo" aria-label="Status message">' . "\n" . ' <h2 class="visually-hidden">Status message</h2>' . "\n" . ' Hello from BigPipe!' . "\n" . ' </div>' . "\n" . '</div>' . "\n",
+          'data' => '<div data-drupal-messages>' . "\n" . ' <div role="region" aria-label="Status message" class="messages messages--status">' . "\n" . ' <div role="status">' . "\n" . ' <h2 class="visually-hidden">Status message</h2>' . "\n" . ' Hello from BigPipe!' . "\n" . ' </div>' . "\n" . ' </div>' . "\n" . ' </div>' . "\n",
           'settings' => NULL,
         ],
       ];
-      $status_messages->embeddedHtmlResponse = '<div data-drupal-messages-fallback class="hidden"></div><div data-drupal-messages>' . "\n" . '  <div role="contentinfo" aria-label="Status message">' . "\n" . '              <h2 class="visually-hidden">Status message</h2>' . "\n" . '              Hello from BigPipe!' . "\n" . '          </div>' . "\n" . '</div>' . "\n";
+      $status_messages->embeddedHtmlResponse = '<div data-drupal-messages-fallback class="hidden"></div><div data-drupal-messages>' . "\n" . '    <div role="contentinfo" aria-label="Status message" class="messages messages--status">' . "\n" . '                  <h2 class="visually-hidden">Status message</h2>' . "\n" . '                    Hello from BigPipe!' . "\n" . '            </div>' . "\n" . '    </div>' . "\n";
     }
 
     // 2. Real-world example of HTML attribute value placeholder: form action.
Index: themes/claro/templates/misc/status-messages.html.twig
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/themes/claro/templates/misc/status-messages.html.twig b/themes/claro/templates/misc/status-messages.html.twig
--- a/themes/claro/templates/misc/status-messages.html.twig	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/themes/claro/templates/misc/status-messages.html.twig	(date 1688139594322)
@@ -39,9 +39,11 @@
         set is_message_with_icon = type in ['error', 'status', 'warning']
       %}
 
-      <div role="contentinfo" aria-labelledby="{{ title_ids[type] }}"{{ attributes.addClass(classes)|without('role', 'aria-label') }}>
+      <div role="region" aria-labelledby="{{ title_ids[type] }}"{{ attributes.addClass(classes)|without('role', 'aria-label') }}>
         {% if type == 'error' %}
           <div role="alert">
+        {% elseif type == 'status' or type == 'warning' %}
+          <div role="status">
         {% endif %}
           {% if is_message_with_title or is_message_with_icon %}
             <div class="messages__header">
@@ -63,7 +65,7 @@
               {{ messages|first }}
             {% endif %}
           </div>
-        {% if type == 'error' %}
+        {% if type in ['error', 'status' , 'warning'] %}
           </div>
         {% endif %}
       </div>
Index: modules/system/templates/status-messages.html.twig
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/system/templates/status-messages.html.twig b/modules/system/templates/status-messages.html.twig
--- a/modules/system/templates/status-messages.html.twig	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/modules/system/templates/status-messages.html.twig	(date 1688139352035)
@@ -23,9 +23,11 @@
 #}
 <div data-drupal-messages>
 {% for type, messages in message_list %}
-  <div role="contentinfo" aria-label="{{ status_headings[type] }}"{{ attributes|without('role', 'aria-label') }}>
+  <div role="region" aria-label="{{ status_headings[type] }}"{{ attributes|without('role', 'aria-label') }}>
     {% if type == 'error' %}
       <div role="alert">
+    {% elseif type == 'status' or type == 'warning' %}
+      <div role="status">
     {% endif %}
     {% if status_headings[type] %}
       <h2 class="visually-hidden">{{ status_headings[type] }}</h2>
@@ -39,7 +41,7 @@
     {% else %}
       {{ messages|first }}
     {% endif %}
-    {% if type == 'error' %}
+    {% if type in ['error', 'status' , 'warning'] %}
       </div>
     {% endif %}
   </div>
Index: modules/big_pipe/tests/src/FunctionalJavascript/BigPipeRegressionTest.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/big_pipe/tests/src/FunctionalJavascript/BigPipeRegressionTest.php b/modules/big_pipe/tests/src/FunctionalJavascript/BigPipeRegressionTest.php
--- a/modules/big_pipe/tests/src/FunctionalJavascript/BigPipeRegressionTest.php	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/modules/big_pipe/tests/src/FunctionalJavascript/BigPipeRegressionTest.php	(date 1688139352032)
@@ -80,7 +80,7 @@
     $this->assertTrue($this->container->get('module_installer')->install(['render_placeholder_message_test'], TRUE), 'Installed modules.');
 
     $this->drupalLogin($this->drupalCreateUser());
-    $messages_markup = '<div role="contentinfo" aria-label="Status message"';
+    $messages_markup = '<div role="region" aria-label="Status message"';
 
     $test_routes = [
       // Messages placeholder rendered first.
Index: profiles/demo_umami/themes/umami/templates/components/messages/status-messages.html.twig
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/profiles/demo_umami/themes/umami/templates/components/messages/status-messages.html.twig b/profiles/demo_umami/themes/umami/templates/components/messages/status-messages.html.twig
--- a/profiles/demo_umami/themes/umami/templates/components/messages/status-messages.html.twig	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/profiles/demo_umami/themes/umami/templates/components/messages/status-messages.html.twig	(date 1688139352046)
@@ -28,10 +28,12 @@
       'messages--' ~ type,
     ]
   %}
-  <div role="contentinfo" aria-label="{{ status_headings[type] }}"{{ attributes.addClass(classes)|without('role', 'aria-label') }}>
+  <div role="region" aria-label="{{ status_headings[type] }}"{{ attributes.addClass(classes)|without('role', 'aria-label') }}>
     <div class="messages__content container">
       {% if type == 'error' %}
         <div role="alert">
+      {% elseif type == 'status' or type == 'warning' %}
+        <div role="status">
       {% endif %}
         {% if status_headings[type] %}
           <h2 class="visually-hidden">{{ status_headings[type] }}</h2>
@@ -45,7 +47,7 @@
         {% else %}
           <span class="messages__item">{{ messages|first }}</span>
         {% endif %}
-      {% if type == 'error' %}
+      {% if type in ['error', 'status' , 'warning'] %}
         </div>
       {% endif %}
     </div>
Index: themes/stable9/templates/media-library/status-messages.html.twig
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/themes/stable9/templates/media-library/status-messages.html.twig b/themes/stable9/templates/media-library/status-messages.html.twig
--- a/themes/stable9/templates/media-library/status-messages.html.twig	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/themes/stable9/templates/media-library/status-messages.html.twig	(date 1688139594412)
@@ -21,9 +21,11 @@
 #}
 <div data-drupal-messages>
 {% for type, messages in message_list %}
-  <div role="contentinfo" aria-label="{{ status_headings[type] }}"{{ attributes|without('role', 'aria-label') }}>
+  <div role="region" aria-label="{{ status_headings[type] }}"{{ attributes|without('role', 'aria-label') }}>
     {% if type == 'error' %}
       <div role="alert">
+    {% elseif type == 'status' or type == 'warning' %}
+      <div role="status">
     {% endif %}
     {% if status_headings[type] %}
       <h2 class="visually-hidden">{{ status_headings[type] }}</h2>
@@ -37,7 +39,7 @@
     {% else %}
       {{ messages|first }}
     {% endif %}
-    {% if type == 'error' %}
+    {% if type in ['error', 'status' , 'warning'] %}
       </div>
     {% endif %}
   </div>
Index: modules/system/tests/themes/test_messages/templates/status-messages.html.twig
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/system/tests/themes/test_messages/templates/status-messages.html.twig b/modules/system/tests/themes/test_messages/templates/status-messages.html.twig
--- a/modules/system/tests/themes/test_messages/templates/status-messages.html.twig	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/modules/system/tests/themes/test_messages/templates/status-messages.html.twig	(date 1688139352040)
@@ -13,9 +13,11 @@
       'messages--' ~ type,
     ]
   %}
-  <div role="contentinfo" aria-label="{{ status_headings[type] }}"{{ attributes.addClass(classes)|without('role', 'aria-label') }}>
+  <div role="region" aria-label="{{ status_headings[type] }}"{{ attributes.addClass(classes)|without('role', 'aria-label') }}>
     {% if type == 'error' %}
       <div role="alert">
+    {% elseif type == 'status' or type == 'warning' %}
+      <div role="status">
     {% endif %}
       {% if status_headings[type] %}
         <h2 class="visually-hidden">{{ status_headings[type] }}</h2>
@@ -29,7 +31,7 @@
       {% else %}
         {{ messages|first }}
       {% endif %}
-    {% if type == 'error' %}
+    {% if type in ['error', 'status' , 'warning'] %}
       </div>
     {% endif %}
   </div>
Index: modules/system/tests/src/Functional/Render/PlaceholderMessageTest.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/system/tests/src/Functional/Render/PlaceholderMessageTest.php b/modules/system/tests/src/Functional/Render/PlaceholderMessageTest.php
--- a/modules/system/tests/src/Functional/Render/PlaceholderMessageTest.php	(revision b83ccdf5ac7b00b033c245725be64239fd61a4d9)
+++ b/modules/system/tests/src/Functional/Render/PlaceholderMessageTest.php	(date 1688139352037)
@@ -28,7 +28,7 @@
    * Tests rendering of message placeholder.
    */
   public function testMessagePlaceholder() {
-    $messages_markup = '<div role="contentinfo" aria-label="Status message"';
+    $messages_markup = '<div role="region" aria-label="Status message"';
 
     $test_routes = [
       // Messages placeholder rendered first.
