diff --git a/src/Plugin/SocialMediaLinks/Iconset/FontAwesome.php b/src/Plugin/SocialMediaLinks/Iconset/FontAwesome.php
index 9704e04..088d411 100755
--- a/src/Plugin/SocialMediaLinks/Iconset/FontAwesome.php
+++ b/src/Plugin/SocialMediaLinks/Iconset/FontAwesome.php
@@ -96,7 +96,7 @@ class FontAwesome extends IconsetBase implements IconsetInterface {
     if ($icon_name == 'envelope' || $icon_name == 'home' || $icon_name == 'rss') {
       $icon = [
         '#type' => 'markup',
-        '#markup' => "<span class='fa fa-$icon_name fa-$style'></span>",
+        '#markup' => "<span class='fa fa-$icon_name fa-$style' aria-hidden=\"true\"></span>",
       ];
     }
     else {
diff --git a/templates/social-media-links-platforms.html.twig b/templates/social-media-links-platforms.html.twig
index a5aecc6..919cf02 100644
--- a/templates/social-media-links-platforms.html.twig
+++ b/templates/social-media-links-platforms.html.twig
@@ -17,15 +17,13 @@
     <li>
       <a class="social-media-link-icon--{{ platform.id }}" href="{{ platform.url | safe_link }}" {{ platform.attributes }} >
         {{ platform.element }}
-      </a>
 
-      {% if appearance.show_name %}
-        {% if appearance.orientation == 'h' %}
-          <br />
+        {% if appearance.show_name and appearance.orientation == 'h' %}
+            <br />
         {% endif %}
 
-        <span><a class="social-media-link--{{ platform.id }}" href="{{ platform.url | safe_link  }}" {{ platform.attributes }}>{{ platform.name }}</a></span>
-      {% endif %}
+        <span class="platform--name{{ appearance.show_name ? '' : ' visually-hidden' }}">{{ platform.name }}</span>
+      </a>
     </li>
   {% endfor %}
 </ul>
