diff --git a/src/Plugin/migrate/process/ConvertBoolean.php b/src/Plugin/migrate/process/ConvertBoolean.php
new file mode 100644
index 0000000..c47988e
--- /dev/null
+++ b/src/Plugin/migrate/process/ConvertBoolean.php
@@ -0,0 +1,41 @@
+<?php
+
+namespace Drupal\migration_tools\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Converts a boolean value to a Drupal boolean value.
+ * Example: TRUE => 1, FALSE => 0.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "convert_boolean"
+ * )
+ *
+ * Example usage:
+ * @code
+ * field_boolean:
+ *   plugin: convert_boolean
+ *   source: boolean
+ * @endcode
+ */
+class ConvertBoolean extends ProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    if (is_string($value) && $value === 'yes' || $value === 'no') {
+      if ($value === 'yes') {
+        $value = TRUE;
+      }
+      else {
+        $value = FALSE;
+      }
+    }
+    return boolval($value);
+  }
+
+}
