diff --git a/src/PluginManager/AiAgentManager.php b/src/PluginManager/AiAgentManager.php
index 8ff423cf9e3ff80a8c18dce576d40958a62608a6..1620f9207902514b1fdd98502520a8004eb5a9e5 100644
--- a/src/PluginManager/AiAgentManager.php
+++ b/src/PluginManager/AiAgentManager.php
@@ -100,20 +100,26 @@ class AiAgentManager extends DefaultPluginManager {
   /**
    * Merge Agent Configurations into this custom plugin system.
    */
-  protected function mergeAgentConfigurations() {
+  protected function mergeAgentConfigurations(): void {
     $plugins = $this->getDefinitions();
     foreach ($plugins as $plugin_id => $plugin) {
       $this->definitions[$plugin_id] = $plugin;
     }
-    $agent_configurations = $this->entityTypeManager->getStorage('ai_agent')->loadMultiple();
-    foreach ($agent_configurations as $id => $entity) {
-      // Modify the plugin definition structure to fit your system.
-      if (!isset($this->definitions[$id])) {
-        $definition['id'] = $id;
-        $definition['label'] = $entity->label();
-        $definition['custom_type'] = 'config';
-        // Add the Config Plugin into our system.
-        $this->definitions[$id] = $definition;
+
+    // If this module is installing, the entity type will not yet exist, so we
+    // don't want to trigger an error.
+    if ($this->entityTypeManager->hasDefinition('ai_agent')) {
+      $agent_configurations = $this->entityTypeManager->getStorage('ai_agent')
+        ->loadMultiple();
+      foreach ($agent_configurations as $id => $entity) {
+        // Modify the plugin definition structure to fit your system.
+        if (!isset($this->definitions[$id])) {
+          $definition['id'] = $id;
+          $definition['label'] = $entity->label();
+          $definition['custom_type'] = 'config';
+          // Add the Config Plugin into our system.
+          $this->definitions[$id] = $definition;
+        }
       }
     }
   }
