diff --git a/src/Service/Fetcher.php b/src/Service/Fetcher.php
index 8913894be7f1934bdbbc7410448e75e8d7ef0fcd..c4af3b96878c6ef3f103c70678152cbbded93e71 100644
--- a/src/Service/Fetcher.php
+++ b/src/Service/Fetcher.php
@@ -10,6 +10,7 @@ use Drupal\Core\Field\FieldDefinitionInterface;
 use Drupal\Core\Logger\LoggerChannelFactoryInterface;
 use Drupal\Core\Messenger\MessengerInterface;
 use Drupal\Core\Render\RendererInterface;
+use Drupal\Core\Url;
 use Drupal\filter\FilterProcessResult;
 use Drupal\filter\Render\FilteredMarkup;
 
@@ -308,6 +309,7 @@ class Fetcher {
     if ($level <= 10 && $source !== NULL) {
       foreach ($data as $fieldname => $values) {
         $this->addProcessedToTextFormatFields($data, $source, $fieldname, $values);
+        $this->convertUriToUrl($data, $source, $fieldname, $values);
         $this->extractEntityReference($data, $source, $fieldname, $level);
       }
     }
@@ -341,6 +343,30 @@ class Fetcher {
     }
   }
 
+  /**
+   * Adds a processed URL for fields with a URI.
+   *
+   * @param array $data
+   *   Array of data fields by reference.
+   * @param object $source
+   *   The source object (paragraph, node, term).
+   * @param string $fieldname
+   *   The field name of what is being processed.
+   * @param string|int|null|array $values
+   *   The values from the entity property or field.  Only operated on if array.
+   */
+  protected function convertUriToUrl(array &$data, object $source, string $fieldname, $values) {
+    if (is_array($values)) {
+      foreach ($values as $index => $instanceValues) {
+        if (!empty($instanceValues['uri'])) {
+          $field = $source->get($fieldname)->get($index);
+          $url = Url::fromUri($field->uri)->toString();
+          $data[$fieldname][$index]['url'] = !empty($url) ? $url : NULL;
+        }
+      }
+    }
+  }
+
   /**
    * Grab the entire referenced entity data.
    *