diff --git a/src/Service/LinkChecker.php b/src/Service/LinkChecker.php
index 5aed40a..9842fb3 100644
--- a/src/Service/LinkChecker.php
+++ b/src/Service/LinkChecker.php
@@ -871,7 +871,7 @@ class LinkChecker {
           // This pattern part is a wildcard.  Automatic match, move on.
           continue;
         }
-        elseif ($pattern_part !== $path_parts[$index]) {
+        elseif (!array_key_exists($index, $path_parts) || $pattern_part !== $path_parts[$index]) {
           // This part did not match, so declare no match and call it off.
           return FALSE;
         }
