diff --git a/src/SubrequestsManager.php b/src/SubrequestsManager.php
index 506d073..8c1b38f 100644
--- a/src/SubrequestsManager.php
+++ b/src/SubrequestsManager.php
@@ -77,7 +77,7 @@ class SubrequestsManager {
         ['master_request' => $master_request]
       );
       $response = $this->httpKernel
-        ->handle($request, HttpKernelInterface::MAIN_REQUEST);
+        ->handle($request, HttpKernelInterface::SUB_REQUEST);
       // Set the Content-ID header in the response.
       $content_id = sprintf('<%s>', $subrequest->requestId);
       $response->headers->set('Content-ID', $content_id);
