diff --git a/src/Service/ViewsBulkOperationsActionProcessor.php b/src/Service/ViewsBulkOperationsActionProcessor.php
index 9d04fe7..69a8de3 100644
--- a/src/Service/ViewsBulkOperationsActionProcessor.php
+++ b/src/Service/ViewsBulkOperationsActionProcessor.php
@@ -419,7 +419,7 @@ class ViewsBulkOperationsActionProcessor implements ViewsBulkOperationsActionPro
     // Check access.
     foreach ($this->queue as $delta => $entity) {
       $accessResult = $this->action->access($entity, $this->currentUser, TRUE);
-      if ($accessResult->isAllowed() === FALSE) {
+      if ($accessResult instanceof AccessResultInterface && $accessResult->isAllowed() === FALSE) {
         $result = [
           'message' => (string) $this->t('Access denied'),
           'type' => 'warning',
