diff --git a/entity_usage.install b/entity_usage.install
index 8871d4e..6a83e47 100644
--- a/entity_usage.install
+++ b/entity_usage.install
@@ -143,15 +143,6 @@ function entity_usage_update_8202(&$sandbox) {
   \Drupal::state()->set('entity_usage_2x_regenerate', TRUE);
 }
 
-/**
- * Re-generate entity_usage statistics.
- */
-function entity_usage_post_update_regenerate_2x(&$sandbox) {
-  // The 1.x to 2.x upgrade is no longer used in the 4.x version and the upgrade
-  // from 1.x to 4.x is not supported. Sites are advised to update to 2.x
-  // version first, before updating to 4.x.
-}
-
 /**
  * Include "target_id_string" also as primary key in schema.
  */
