diff --git a/paragraphs.links.contextual.yml b/paragraphs.links.contextual.yml
new file mode 100644
index 0000000..a989763
--- /dev/null
+++ b/paragraphs.links.contextual.yml
@@ -0,0 +1,4 @@
+paragraphs.edit_paragraph:
+  title: 'Edit'
+  route_name: paragraphs.edit_paragraph
+  group: paragraph
diff --git a/paragraphs.routing.yml b/paragraphs.routing.yml
index ee8b03f..f8d5166 100644
--- a/paragraphs.routing.yml
+++ b/paragraphs.routing.yml
@@ -37,3 +37,11 @@ paragraphs.settings:
     _title: 'Paragraphs settings'
   requirements:
     _permission: 'administer paragraphs settings'
+
+paragraphs.edit_paragraph:
+  path: '/paragraph/{paragraph}/edit'
+  defaults:
+    _controller: '\Drupal\paragraphs\Controller\ParagraphsContextualLinkController::goToEdit'
+  requirements:
+    _permission: 'access content'
+    node: \d+
diff --git a/src/Controller/ParagraphsContextualLinkController.php b/src/Controller/ParagraphsContextualLinkController.php
new file mode 100644
index 0000000..97150e1
--- /dev/null
+++ b/src/Controller/ParagraphsContextualLinkController.php
@@ -0,0 +1,58 @@
+<?php
+
+
+namespace Drupal\paragraphs\Controller;
+
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Routing\IgnoreDestinationRedirectResponse;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\Core\Url;
+
+class ParagraphsContextualLinkController extends ControllerBase {
+  public function goToEdit($paragraph) {
+    $parents = [];
+    $paragraph_object = $this->entityTypeManager()->getStorage('paragraph')->load($paragraph);
+
+    $ancestor = self::get_parents($paragraph_object, $parents);
+    if (!empty($ancestor)) {
+      $options['query'] = [];
+      $options['query']['edit_para'][] = $paragraph;
+      $options['fragment'] = 'p' . $paragraph;
+      foreach ($parents as $parent) {
+        $options['query']['edit_para'][] = $parent;
+      }
+
+      return new IgnoreDestinationRedirectResponse(Url::fromRoute('entity.node.edit_form', $ancestor, $options)->setAbsolute()->toString());
+    }
+    else {
+      throw new \Exception($this->t('Could not find edit screen for paragraph %id', ['%id' => $paragraph]));
+    }
+  }
+
+  /**
+   * Adds ids of any containing paragraphs to parent_ids and returns [type => id] of the entity that contains them.
+   *
+   * @param \Drupal\paragraphs\Entity\Paragraph $paragraph
+   *   The paragraph to find the parents of.
+   * @param array $paragraph_ids
+   *   An array to hold ancestor paragraph ids.
+   *
+   * @return array
+   *   The id of the non-paragraph entity where the paragraph lives keyed on entity type.
+   */
+  public static function get_parents(Paragraph $paragraph, array &$paragraph_ids=[]) {
+    $ancestor = [];
+
+    $parent = $paragraph->getParentEntity();
+    if ($parent->getEntityType()->id() === 'paragraph') {
+      $paragraph_ids[] = $parent->id();
+      $node_id = self::get_parents($parent, $paragraph_ids);
+    }
+    else {
+      $ancestor[$parent->getEntityType()->id()] = $parent->id();
+    }
+    return $ancestor;
+  }
+
+}
diff --git a/src/ParagraphViewBuilder.php b/src/ParagraphViewBuilder.php
index 87b45eb..891fec7 100644
--- a/src/ParagraphViewBuilder.php
+++ b/src/ParagraphViewBuilder.php
@@ -5,6 +5,8 @@ namespace Drupal\paragraphs;
 use Drupal\Core\Entity\Entity\EntityViewDisplay;
 use Drupal\Core\Entity\EntityViewBuilder;
 use Drupal\Core\Render\Element;
+use Drupal\paragraphs\Controller\ParagraphsContextualLinkController;
+use Drupal\paragraphs\Entity\Paragraph;
 
 /**
  * Render controller for paragraphs.
@@ -38,6 +40,17 @@ class ParagraphViewBuilder extends EntityViewBuilder {
         $plugin_value->view($build_list[$key], $build['#paragraph'], $display, $build['#view_mode']);
       }
       $build_list[$key]['#attached']['library'][] = 'paragraphs/drupal.paragraphs.unpublished';
+
+      /** @var Paragraph $paragraph */
+      $paragraph = $build_list[$key]['#paragraph'];
+      $parent = $paragraph->getParentEntity();
+      if (ParagraphsContextualLinkController::get_parents($paragraph)) {
+        $build_list[$key]['#contextual_links']['paragraph'] = [
+            'route_parameters' => [
+                'paragraph' => $paragraph->id(),
+            ]
+        ];
+      }
     }
 
     return $build_list;
diff --git a/src/Plugin/Field/FieldWidget/ParagraphsWidget.php b/src/Plugin/Field/FieldWidget/ParagraphsWidget.php
index 001e9e0..1278089 100644
--- a/src/Plugin/Field/FieldWidget/ParagraphsWidget.php
+++ b/src/Plugin/Field/FieldWidget/ParagraphsWidget.php
@@ -379,8 +379,12 @@ class ParagraphsWidget extends WidgetBase {
     elseif (isset($items[$delta]->entity)) {
       $paragraphs_entity = $items[$delta]->entity;
 
+      // If there's an 'edit_para' parameters, open listed paragraphs.
+      if (isset($_GET['edit_para']) && in_array($paragraphs_entity->id(),  $_GET['edit_para'])) {
+        $item_mode = 'edit';
+      }
       // We don't have a widget state yet, get from selector settings.
-      if (!isset($widget_state['paragraphs'][$delta]['mode'])) {
+      elseif (!isset($widget_state['paragraphs'][$delta]['mode'])) {
 
         if ($default_edit_mode == 'open' || $widget_state['items_count'] < $this->getSetting('closed_mode_threshold')) {
           $item_mode = 'edit';
@@ -518,7 +522,8 @@ class ParagraphsWidget extends WidgetBase {
         ),
       );
 
-      $element['#prefix'] = '<div id="' . $wrapper_id . '">';
+      // Create an anchor based on the paragraph id.
+      $element['#prefix'] = '<span id="p' . $paragraphs_entity->id() . '"></span><div id="' . $wrapper_id . '">';
       $element['#suffix'] = '</div>';
 
       // Create top section structure with all needed subsections.
