diff --git a/core/modules/menu_link_content/src/MenuLinkContentAccessControlHandler.php b/core/modules/menu_link_content/src/MenuLinkContentAccessControlHandler.php
index 496d87fe1b..a0013179ca 100644
--- a/core/modules/menu_link_content/src/MenuLinkContentAccessControlHandler.php
+++ b/core/modules/menu_link_content/src/MenuLinkContentAccessControlHandler.php
@@ -78,6 +78,17 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
             ->addCacheableDependency($entity);
         }
 
+        $allowed_paths = [
+          '/find-locations',
+          '/health-care/apply/application'
+        ];
+
+        if (in_array($url_object->toString(), $allowed_paths)) {
+          return AccessResult::allowed()
+            ->cachePerPermissions()
+            ->addCacheableDependency($entity);
+        }
+
       case 'update':
         if (!$account->hasPermission('administer menu')) {
           return AccessResult::neutral("The 'administer menu' permission is required.")->cachePerPermissions();
